content/ADR-253-light-sources.md

```diff
@@ -0,0 +1,117 @@
+
+# Context
+
+Up until now, all light in Decentraland came from a single rigid source (the sun or moon). Enabling creators to create their own light sources enables a lot of creative possibilities. We see the control of light as important for the following:
+
+- Creating ambience and enhance the architecture of a scene
+- Flashy effects for music festivals, fashion week, and other events of high visual impact
+- Light as a visual cue: guide the player towards what’s important in a scene by shining a light on it. Signal a change in what’s going on in the scene by switching the lights to a different intensity or color.
+- Darkness as a mechanic for spooky games
+- Flashlights as a mechanic: the act of having to point a light to reveal what’s there can be a whole game mechanic.
+
+To be able to play with darkness, we'll also need to provide controls to be able to disable the global default light sources and play with that.
+
+Note: Some GLTF models come with lights packaged as nodes inside the structure of the model. We should ignore those completely. The only lights we will be rendering are the ones defined via code using the LightSource component.
+
+## Component description
+
+We should create a `LightSource` component that, when added to an entity, tells the engine to shine a light from that entity’s Transform position.
+
+A Type field will let you chose between _Spot_ and _Point_ light. We believe these two types of lights are enough for now, the component could be expanded in the future if we want to include other types.
+
+A _Spot_ light projects a cone of concentrated light in a single direction, originating from a single point in the scene. It's ideal for shows and theatric effects, it can also be used to shine a light on something with a functional intention.
+
+A _Point_ light expands in all directions from a point in the scene, illuminating the space in a more subtle way. The point of origin is often not easy to pinpoint for the player.
+
+The following fields will be available on both types of light:
+
+- Color: _Color4_ The color of the light
+- Intensity: _number_ The luminosity value of the light, from 0 to 1.
+- Range: _number_ The maximum distance that can be affected by the light source.
+- Active: _boolean_ Determines if the light is currently on or not.
+
+In lights of type _Spot_, we will also include:
+
+- Inner angle: _number_ This angle, measured from the direction vertex, defines a cone where the light has full intensity. Max 180, can’t be more than outer angle.
+- Outer angle: _number_ This angle, measured from the direction vertex, defines a cone where the light has an effect, the intensity decreases farther away from the inner cone. Max 180.
+
+The `Active` flag lets creators easily turn a light source on or off. We could otherwise achieve the same by setting intensity to 0, but offering a switch makes it easier to retain the prior configuration.
+
+## Shadows
```

Comment:
Regarding shadows, I suggest expanding this section to address several important aspects:

1. **Shadow Control**
Instead of a simple boolean, consider these key properties:
- `shadows: boolean` - Whether the light casts shadows
- Default values should differ by light type:
  * Point/Spot lights: `false` (performance consideration)
  * Global/Directional light: `true` (essential for scene depth)

2. **Implementation Considerations**
Add a note that shadow rendering may vary by platform:
```typescript
// Even when shadows = true, the engine may:
// - Not display shadows at all
// - Show shadows for limited number of lights
// - Vary shadow quality
// Based on:
// - Platform capabilities
// - User settings
// - Performance requirements
```

3. **Priority System**
Consider adding guidance for implementations about shadow priority:
- Global/Directional light shadows take precedence
- Point/Spot lights could use distance/illuminance for priority

This approach:
- Gives developers clear expectations
- Allows implementations to optimize performance
- Maintains flexibility for different platforms
- Provides predictable behavior across scenes

See our protocol implementation for a working example of these concepts.

Reply:
We agree that it's better to have shadows as a simple boolean, and let each engine determine things like shadow resolution, priorities, etc.
I changed this in the document